app/assets/javascripts/app/estimate_products_table.js

```diff
@@ -78,8 +79,8 @@ $(function () {
       var row = $(this).closest("tr");
       var destroyField = row.find(".destroy-field");
 
-      if (row.data("estimate_detail_id")) {
-        destroyField.val("1");
+      if (destroyField.length) {
+        destroyField.val(true);
```

Comment:
This also sets `_destroy` for a new row, which is unnecessary. Instead of just hiding them, we can remove them if they were never in the DB